src/sinks/http.rs

```diff
@@ -232,9 +231,17 @@ impl HttpSink for HttpSinkConfig {
             Encoding::Text => "text/plain",
             Encoding::Ndjson => "application/x-ndjson",
             Encoding::Json => {
-                body.insert(0, b'[');
-                body.pop(); // remove trailing comma from last record
-                body.push(b']');
+                // TODO(https://github.com/vectordotdev/vector/issues/11253):
+                // Prepend before building a request body to eliminate the
+                // additional copy here.
+                let message = body.split();
+                body.put_u8(b'[');
+                if !message.is_empty() {
+                    body.unsplit(message);
+                    // remove trailing comma from last record
+                    body.truncate(body.len() - 1);
```

Comment:
Disappointing that `BytesMut` doesn't have a `pop`. It would almost be worth having a common function for this, thought I suspect the mental overhead of figuring out that function is larger than just writing it out.

Reply:
Oh yeah, felt the same way here 😄